src/ngraph_builder.cc

```diff
@@ -3264,35 +3264,54 @@ static Status TranslateSplitVOp(
     const Node* op, const std::vector<const Tensor*>& static_input_map,
     Builder::OpMap& ng_op_map) {
   shared_ptr<ng::Node> ng_input, ng_length, ng_split_dim;
-  TF_RETURN_IF_ERROR(
-      GetInputNodes(ng_op_map, op, &ng_input, &ng_length, &ng_split_dim));
+  TF_RETURN_IF_ERROR(GetInputNode(ng_op_map, op, 0, &ng_input));
 
   std::vector<int> lengths;
   TF_RETURN_IF_ERROR(GetStaticInputVector(op, 1, static_input_map, &lengths));
 
+  int length = 0;
+  int shape_length = 0;
+  int idx;
+  for (int i = 0; i < lengths.size(); ++i) {
+    if (lengths[i] != -1) {
+      length += lengths[i];
+    } else {
+      idx = i;
+    }
+  }
+
   ng::Shape shape = ng_input->get_shape();
   int rank = shape.size();
-  std::vector<size_t> lower;
-  std::vector<size_t> upper;
-
-  for (int i = 0; i < rank; ++i) {
-    lower.push_back(0);
-    upper.push_back(shape[i]);
+  for (int i = 0; i < shape.size(); ++i) {
+    shape_length += shape[i];
+  }
+  if (shape_length > length) {
+    lengths[idx] = shape_length - length;
   }
 
+  std::vector<size_t> lower(rank, 0);
+  std::vector<size_t> upper(shape);
+
   std::vector<int> split_dim_vec;
   TF_RETURN_IF_ERROR(
       GetStaticInputVector(op, 2, static_input_map, &split_dim_vec));
-  int split_dim = split_dim_vec[0];
+
+  int split_dim = split_dim_vec[0] + (split_dim_vec[0] < 0 ? (int64)rank : 0);
+
   int cursor = 0;
 
-  for (int i = 0; i < lengths.size(); ++i) {
-    lower[split_dim] = cursor;
-    cursor += lengths[i];
-    upper[split_dim] = cursor;
-    SaveNgOp(ng_op_map, op->name(),
-             make_shared<ng::op::Slice>(ng_input, lower, upper));
+  if (lengths.size() != 1) {
+    for (int i = 0; i < lengths.size(); ++i) {
+      cout << "Printing lengths vector:" << lengths[i] << endl;
+      lower[split_dim] = cursor;
+      upper[split_dim] = cursor;
```

Comment:
```
lower[split_dim] = cursor;
upper[split_dim] = cursor;
```

Both are being set to exact same value, hence we see `0`-dimension

please note of the original code, which does the cursor updating in between setting lower and upper